test_data/asm/permutations/link_merging.asm

```diff
@@ -29,7 +29,7 @@ machine Main {
 
     // these are merged into 1 link
     instr add X, Y -> Z link => Z = submachine.add(X, Y);
-    instr sub_with_add X, Y -> Z link => Y = submachine.add(X, Z);
+    instr sub_with_add X, Y -> Z link => X = submachine.add(Y, Z);
```

Comment:
```suggestion
    instr sub_with_add X, Y -> Z link => Z = submachine.add(X, Y);
```
Shouldn't it be this?

Reply:
`instr sub_with_add X, Y -> Z` means that we want X - Y = Z (X = Y + Z). So, shouldn't we use `link => X = add(Y, Z)`

Reply:
It is not `instr add X, Y -> Z`, it is ìnstr sub_with_add`

Reply:
ah right, I missed the sub part, agree with your PR